src/OpenTelemetry.Contrib.Extensions.AWSXRay/Instrumentation/AWSTracingPipelineHandler.cs

```diff
@@ -0,0 +1,238 @@
+
+        private static readonly AWSXRayPropagator AwsPropagator = new AWSXRayPropagator();
+        private static readonly Action<IDictionary<string, string>, string, string> Setter = (carrier, name, value) =>
+        {
+            carrier[name] = value;
+        };
+
+        private static readonly Dictionary<string, string> ServiceParameterMap = new Dictionary<string, string>()
+        {
+            { "DynamoDBv2", "TableName" },
+            { "SQS", "QueueUrl" },
+        };
+
+        private static readonly Dictionary<string, string> ParameterAttributeMap = new Dictionary<string, string>()
+        {
+            { "TableName", AWSSemanticConventions.AttributeAWSDynamoTableName },
+            { "QueueUrl", AWSSemanticConventions.AttributeAWSSQSQueueUrl },
+        };
+
+        private static readonly ActivitySource AWSSDKActivitySource = new ActivitySource(ActivitySourceName);
+
+        public override void InvokeSync(IExecutionContext executionContext)
+        {
+            var activity = this.ProcessBeginRequest(executionContext);
+            try
+            {
+                base.InvokeSync(executionContext);
+            }
+            catch (Exception ex)
+            {
+                if (activity != null)
+                {
+                    this.ProcessException(activity, ex);
+                }
+            }
+            finally
+            {
+                if (activity != null)
+                {
+                    this.ProcessEndRequest(executionContext, activity);
+                }
+            }
+        }
+
+        public override async Task<T> InvokeAsync<T>(IExecutionContext executionContext)
+        {
+            T ret = null;
+
+            var activity = this.ProcessBeginRequest(executionContext);
+            try
+            {
+                ret = await base.InvokeAsync<T>(executionContext).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                if (activity != null)
+                {
+                    this.ProcessException(activity, ex);
+                }
+            }
+            finally
+            {
+                if (activity != null)
+                {
+                    this.ProcessEndRequest(executionContext, activity);
+                }
+            }
+
+            return ret;
+        }
+
+        private Activity ProcessBeginRequest(IExecutionContext executionContext)
+        {
+            Activity activity = null;
+
+            var requestContext = executionContext.RequestContext;
+            var service = this.GetAWSServiceName(requestContext);
+            var operation = this.GetAWSOperationName(requestContext);
+
+            activity = AWSSDKActivitySource.StartActivity(service + "." + operation, ActivityKind.Client);
+
+            if (activity == null)
+            {
+                return null;
+            }
+
+            activity.AddTag(AWSSemanticConventions.AttributeAWSServiceName, service);
```

Comment:
Replace AddTag for SetTag (in this and all other places)

Reply:
Done